client/src/views/Viewer.vue

```diff
@@ -411,18 +423,27 @@ export default {
     /* END TODO */
     async loadDataset(datasetId) {
       var { data: dataset } = await this.girderRest.get(`folder/${datasetId}`);
-      if (!dataset || !dataset.meta || !dataset.meta.viame) {
-        return null;
-      }
-      this.dataset = dataset;
+      this.dataset = dataset || null;
     },
     async loadDetections() {
-      var { data: detections } = await this.girderRest.get("viame_detection", {
-        params: { folderId: this.dataset._id }
-      });
-      this.detections = detections.map(detection => {
-        return Object.freeze(detection);
-      });
+      let detections = [];
+      try {
+        const { data } = await this.girderRest.get("viame_detection", {
+          params: { folderId: this.dataset._id }
+        });
+
+        detections = data
+          ? data.map(detection => Object.freeze(detection))
```

Comment:
I know it was never commented before, but using object.freeze to protect objects from reactivity like this really deserves some explanation.  Some developer will come along later and wonder what was going on.

Reply:
I'm not sure why it's being done either, it was there before I came onto the project. I think it's deserving of it's own PR though.